src/Swarm/Game/Step.hs

```diff
@@ -736,7 +755,7 @@ execConst c vs s k = do
         return $ Waiting (time + d) (Out VUnit s k)
       _ -> badConst
     Selfdestruct -> do
-      destroyIfNotBase
+      destroyIfNotBase $ Just AttemptSelfDestructBase
```

Comment:
Note this is currently only possible in creative mode, since it is not possible to craft a `detonator`.  That's fine though.